src/main/java/pl/widulinski/scrapp/urls/URLService.java

```diff
@@ -0,0 +1,130 @@
+package pl.widulinski.scrapp.urls;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.springframework.stereotype.Service;
+import pl.widulinski.scrapp.ScrappedWebElement.ScrappedWebElement;
+import pl.widulinski.scrapp.createExcel.CreateExcel;
+import pl.widulinski.scrapp.enums.Categories;
+import pl.widulinski.scrapp.webDataToScrap.DataToScrap;
+import pl.widulinski.scrapp.webDataToScrap.DataToScrapRepository;
+import pl.widulinski.scrapp.webDriver.BrowserPicker;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+
+@Service
+@Data
+@Slf4j
+public class URLService extends BrowserPicker {
+
+    private int quantityOfPages = 0;
+
+    private CreateExcel createExcel;
+
+    private  DataToScrapRepository dataToScrapRepository;
```

Comment:
przez `@Data` musisz robic nie-finalne zmienne
`@Data` = `@Getter`, `@Setter`, `@RequiredArgsContrcutor`, `@EqualsAndHashcode` `@ToString`
W mojej ocenie w klasie serwisowej powinieneś mieć zmienne finalne + `@RequiredArgsContrcutor` i tyle

Nie wiem czy gdzies robisz `urlService.getCreateExcel` albo `getDataToScrapRepostiry`. Nie powinieneś ale jeśli robisz to dodaj jeszcez po prostu `@Getter` 

`@Data` sugeruje ze to klasa modelowa a ewidentnie ją nie jest

Reply:
ok jasne. @Data wywalone, stworzyłem tylko setter do quantityOfPages (wykorzystuje go w testach)